### DISCUSSION TITLE
Stream and Struct Blocks' Keyboard shortcuts

### DISCUSSION THREAD

**Discussion by hazho** [wagtail/wagtail#7642] [Ideas]
I think, in this era (with lots of competitors in the market) we should provide the basic features of editing for edit_handler, example: the keyboard shortcuts..!
----------------------
discussions to take dicision about: 
- what actions(events) should be represented a shortcut for.
- how to (back-end and front-end) control the focused block.
- providing modifiable interface (keys to be changed by developer/superuser).
- and ...any other related discussions.

**Comment by gasman** [comment]
Can you give some specific examples of keyboard shortcuts that you feel are missing? StreamField and edit_handlers are completely separate areas of Wagtail, so it's hard to tell what you're referring to here.

**Reply by hazho** [reply]
Basically True, but we need to create shortcuts for both, Examples:
**StreamField:**
 

-    Copy a block (struct, list or any type of blocks) by Ctrl+C when the specific block is in Focussed mode.
-    Past the Copied block in the focused Acceptable parent to the copied block type.
-    ...etc.

**None StreamField parts of EditHandler:**
  

-   Saving the page (revision) by Ctrl+S no matter where is the focussed area in the page (not looking after any focused objects in the page). 
-    Undo and Redo of any Chnages, this can be achieved by the browser's local storage events or the OS's Events storage(which is I'm not sure how to do that).
- ...etc.
